File: pinyougou-web/pinyougou-user-web/src/main/webapp/js/controller/lickedController.js
app.controller('lickedController', function($scope, $controller, $timeout, baseService) {

    $controller('indexController', {$scope: $scope});

    $scope.findAllApplication = function () {
        baseService.sendGet("/licked/findLickers").then(function (response) {
            if (response.data.licker) {
                $scope.myLicker = response.data.licker;
            }
            $scope.applicationList = response.data.lilckerList;
        })
    };

    $scope.getLickedMsg = function () {
        baseService.sendGet("/licked/getLickedMsg").then(function (response) {
            var msg = "";
            if (response.data != null && response.data.length > 0) {
                for (var i = 0; i < response.data.length; i++) {
                    msg += JSON.stringify(response.data[i]) + "\n";
                }
                alert(msg);
            }
        })
    };


    $scope.acceptLicker = function (lickerId) {
        baseService.sendGet("/licked/acceptLicker?lickerId=" + lickerId).then(function (response) {
            if (response.data) {
                $scope.findAllApplication();
            } else {
                alert("操作失败")
            }
        })
    };

    $scope.refuseLicker = function (lickerId) {
        baseService.sendGet("/licked/refuseLicker?lickerId=" + lickerId).then(function (response) {
            if (response.data) {
                $scope.findAllApplication();
            } else {
                alert("操作失败")
            }
        })
    };

    $scope.deleteLicker = function (lickerId) {
        baseService.sendGet("/licked/deleteLicker?lickerId=" + lickerId).then(function (response) {
            $scope.findAllApplication();
        })
    }
});

File: pinyougou-web/pinyougou-user-web/src/main/java/com/pinyougou/user/controller/lickedController.java
package com.pinyougou.user.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.service.LickService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import javax.servlet.http.HttpServletRequest;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/licked")
public class lickedController {

    @Autowired
    private HttpServletRequest request;

    @Reference(timeout = 100000)
    private LickService lickService;

    @GetMapping("/findLickers")
    public Map<String, Object> findLickers() {
        String userId = request.getRemoteUser();
        Map<String, Object> data = new HashMap<>();

        List<String> lickerList = lickService.findLickers(userId);
        data.put("lilckerList", lickerList);

        String lickerId = lickService.findMyLicker(userId);
        if (lickerId != null && lickerId != "") {
            data.put("licker", lickerId);
        }

        return data;
    }

    @GetMapping("/acceptLicker")
    public boolean acceptLicker(String lickerId) {
        try {
            String lickedId = request.getRemoteUser();

            lickService.acceptLicker(lickerId, lickedId);

            return true;
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }

    @GetMapping("/refuseLicker")
    public boolean refuseLicker(String lickerId) {
        try {
            String lickedId = request.getRemoteUser();

            lickService.refuseLicker(lickerId, lickedId);

            return true;
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }

    @GetMapping("/deleteLicker")
    public boolean deleteLicker(String lickerId) {
        try {
            String lickedId = request.getRemoteUser();

            lickService.brokeUp(lickerId, lickedId);

            return true;
        } catch (Exception e) {
            e.printStackTrace();
            return false;
        }
    }

    @GetMapping("/getLickedMsg")
    public List<String> getLickedMsg() {
        String lickedId = request.getRemoteUser();

        List<String> lickedMsgList = lickService.getLickedMsg(lickedId);

        if (lickedMsgList == null || lickedMsgList.size() == 0) {
            lickedMsgList = new ArrayList<>();
        }

        return lickedMsgList;
    }
}
